modules/history/src/views/full/MessageViewer.jsx

```diff
@@ -164,19 +185,38 @@ export class MessageViewer extends React.Component {
             this.state.inspectorIsShown ? style['message-list-partial'] : style['message-list-full']
           )}
         >
-          <MessagesHeader conversation={this.props.conversation} messageGroups={this.props.messageGroups} />
-          {this.props.messageGroups &&
-            this.props.messageGroups.map(group => {
-              return (
-                <MessageGroup
-                  key={group[0].id}
-                  messages={group}
-                  focusMessage={m => this.setState({ currentlyFocusedMessage: m, inspectorIsShown: true })}
-                />
-              )
-            })}
+          <MessagesHeader conversation={this.state.currentConversation} messageGroups={this.props.messageGroups} />
+          <MessagesTaskBar
+            ref="taskBar"
+            useAsFilter={!this.state.areMessagesSelected}
+            flag={() => this.flagSelectedMessages()}
+            unflag={() => this.unflagSelectedMessages()}
```

Comment:
Unless you pass parameters, no need of an anonymous function: flag={this.flagSelectedMessages}